extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php

```diff
@@ -34,11 +48,68 @@ static public function onImageServingCollectImages( &$imageNamesArray, $articleT
 	 * @param $parser
 	 * @param $frame
 	 */
-	static public function onParserTagHooksBeforeInvoke( $name, $marker, $content, $attributes, $parser, $frame ) {
-		if ( $name === self::PARSER_TAG_GALLERY ) {
-			\Wikia\PortableInfobox\Helpers\PortableInfoboxDataBag::getInstance()->setGallery( $marker, $content );
+	static public function onParserTagHooksBeforeInvoke($name, $marker, $content, $attributes, $parser, $frame) {
+		if ($name === self::PARSER_TAG_GALLERY) {
+			\Wikia\PortableInfobox\Helpers\PortableInfoboxDataBag::getInstance()->setGallery($marker, $content);
+		}
+
+		return true;
+	}
+
+	/**
+	 * renders portable infobox builder instead of edit page while creating new template
+	 *
+	 * @param EditPageLayoutController $editPageContext
+	 * @return bool
+	 */
+	static public function onEditPageLayoutExecute( $editPageContext ) {
+		global $wgRequest;
```

Comment:
You could acquire request from $editPageContext->getRequest(), have you tried it?